src/controllers/hooks.js

```diff
@@ -282,6 +299,19 @@ function HooksController(lambdaContext) {
 
     // discard the site candidate if the site exists in sites db with deliveryType=aem_edge
     if (site && site.getDeliveryType() === DELIVERY_TYPES.AEM_EDGE) {
+      // for existing site with empty hlxConfig, update it now
+      // todo: temporary to backfill hlx config for existing sites
+      /* c8 ignore start */
+      if (
+        source === SITE_CANDIDATE_SOURCES.CDN
+        && isObject(hlxConfig) && Object.keys(hlxConfig).length > 0
+        && (!isObject(site.getHlxConfig()) || Object.keys(site.getHlxConfig()).length === 0)
```

Comment:
idea: we can introduce a `isNonEmptyObject`method to the shared

Reply:
@ekremney here you go https://github.com/adobe/spacecat-shared/pull/280